### PR TITLE
Fixed import-db

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- there was a bug in the import-db process if we have used the old version of DBs that did not contain the `ActiveDB` marker. The versions <=1.3.36 did not care about it as it would have re-written anyway that marker and the snapshotting process worked as expected. With the rc/2022-july branch, a new condition appeared as to write the marker only on epoch 0;
- some prints regarding the wrong epoch data were not in line with the rest of the project, refactored the prints for better debugging experience.
  
## Proposed Changes
- add a composed condition in which, we take for granted in `import-db` process that the DB is correctly constructed as to reuse the old trie databases.

## Testing procedure
- `import-db` process should work as expected, without ERRORs or WARNs;
- internal testing scenarios should work.

### CODE REVIEW

File: epochStart/metachain/epochStartData.go [hunk 1]
```diff
@@ -106,21 +107,21 @@ func (e *epochStartData) VerifyEpochStartDataForMetablock(metaBlock *block.MetaB
...
 }
 
-func displayEpochStartData(startData *block.EpochStart) {
+func displayEpochStartData(mode string, hash []byte, startData *block.EpochStart) {
+	log.Warn(mode+" epoch start data",
+		"hash", hash, "startData", spew.Sdump(startData))
+
 	for _, shardData := range startData.LastFinalizedHeaders {
-		log.Debug("epoch start shard data",
+		log.Warn("epoch start shard data",
```
Comment (bogdan-rosianu) [inline epochStart/metachain/epochStartData.go:124]: change back to DEBUG ?

File: state/accountsDB.go [hunk 2]
```diff
@@ -156,12 +156,16 @@ func checkArgsAccountsDB(args ArgsAccountsDB) error {
...
-func startSnapshotAfterRestart(adb AccountsAdapter, tsm common.StorageManager) {
+func startSnapshotAfterRestart(adb AccountsAdapter, args ArgsAccountsDB) {
+	tsm := args.Trie.GetStorageManager()
 	epoch, err := tsm.GetLatestStorageEpoch()
 	if err != nil {
 		log.Error("could not get latest storage epoch")
 	}
-	if epoch == 0 && err == nil {
+	putActiveDBMarker := epoch == 0 && err == nil
+	putActiveDBMarker = putActiveDBMarker || args.ProcessingMode == common.ImportDb
```
Comment (bogdan-rosianu) [inline state/accountsDB.go:166]: might add a new variable. for example `isInImportDBMode` and then check if either of them is true

**Review by BeniaminDrasovean — DISMISSED** [review]

File: epochStart/metachain/epochStartData.go (hunk 1 above)
Comment (iulianpascalau) [reply epochStart/metachain/epochStartData.go:124]: no, as the message that indicate a serious problem is on WARN level, the details are on debug level

File: state/accountsDB.go (hunk 2 above)
Comment (iulianpascalau) [reply state/accountsDB.go:166]: done

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by popenta — APPROVED** [review]
System test passed

**Pull request merged**